mbtiles/mp.py

```diff
@@ -0,0 +1,52 @@
+# multiprocessing Pool implementation
+
+from multiprocessing import Pool
+import warnings
+
+from mbtiles.compat import zip_longest
+from mbtiles.worker import init_worker, process_tile
+
+BATCH_SIZE = 100
+
+
+class MbtilesDeprecationWarning(FutureWarning):
+    """For rio-mbtiles deprecations"""
+
+
+warnings.warn(
+    "The multiprocessing.Pool implementation will be removed in rio-mbtiles 2.0.0.",
+    MbtilesDeprecationWarning,
+    stacklevel=2,
+)
+
+
+def grouper(iterable, n, fillvalue=None):
+    "Collect data into fixed-length chunks or blocks"
+    # grouper('ABCDEFG', 3, 'x') --> ABC DEF Gxx"
+    args = [iter(iterable)] * n
+    return zip_longest(*args, fillvalue=fillvalue)
+
+
+def process_tiles(
+    conn,
+    tiles,
+    insert_results,
+    num_workers=None,
+    inputfile=None,
+    base_kwds=None,
+    resampling=None,
+    img_ext=None,
+    image_dump=None,
+):
+    pool = Pool(
+        num_workers, init_worker, (inputfile, base_kwds, resampling), BATCH_SIZE
```

Comment:
I'm curious about the batching here, if you don't mind sharing a bit of the motivation.

Reply:
It's a hedge against resource leaks, but now that you mention it, this could be too low.